client/js/shout.js

```diff
@@ -682,18 +658,47 @@ $(function() {
 			"quit",
 			"nick",
 			"mode",
+			"whois"
 		];
 		if ($.inArray(type, ignore) !== -1){
 			return;
 		}
 
-		var badge = button.find(".badge");
-		if (badge.length !== 0) {
-			var i = (badge.data("count") || 0) + 1;
-			badge.data("count", i);
-			badge.html(i > 999 ? (i / 1000).toFixed(1) + "k" : i);
-			if (highlight || isQuery) {
-				badge.addClass("highlight");
+		var isQuery = button.hasClass("query");
+		var highlight = type.contains("highlight");
+		var message = type.contains("message");
+		var settings = $.cookie("settings") || {};
```

Comment:
Can't you access `options` variable directly here? I know I'm nitpicking on the existing code, but might as well improve it.